### PR TITLE
more changes

### PR DESCRIPTION
**Pull request by obrakmann**
sorry, I just noticed this while looking over the changes, and since it involved touching the lua script I thought I'd do it myself.

### CODE REVIEW

File: mods/cnc/maps/gdi05a/gdi05a.lua [hunk 1]
```diff
@@ -117,8 +117,8 @@ Grd3Action = function()
 	end
 end
 
-DiscoverGdiBase = function()
-	if baseDiscovered then
+DiscoverGdiBase = function(actor, discoverer)
```
Comment (penev92) [inline mods/cnc/maps/gdi05a/gdi05a.lua:120]: Are you sure this should be "actor" and not "player" or "discovered" ?

**Comment by penev92** [comment]
Hm, wonder what happened to my earlier comment...

OK, so I could take the the Lua script changes. Those are mission-specific and are your call (although I don't see the point).
But the fact that gdi05a doesn't use the `Player discoverer` argument doesn't mean that others won't need it.

File: mods/cnc/maps/gdi05a/gdi05a.lua (hunk 1 above)
Comment (obrakmann) [inline mods/cnc/maps/gdi05a/gdi05a.lua:120]: doesn't matter, I'm not using that variable at all

**Comment by obrakmann** [comment]
No, I'm afraid you haven't understood that right. The method signature in `TriggerGlobal.cs` has nothing to do with what the callback will look like. The callback will get whatever arguments you give it in `ScriptTriggers.cs` (line 335 for the `OnDiscovered` trigger).  In `TriggerGlobal` you just register the trigger, and you don't need the `discoverer` argument for that, it's just not used (it's just a single line of code after all).

Edit: and the lua script changes are just because I didn't realize earlier that I had to check _which_ player actually triggered the `OnDiscovered`. It could have been Nod, too (theoretically).

And now really sleep

**Comment by penev92** [comment]
You have tested that both triggers still work, right?
Because I will merge this now and test tomorrow.

**Comment by penev92** [comment]
Hello? https://github.com/OpenRA/OpenRA/pull/7463
Judging by the fact that his Lua file has a `GdiBaseDiscoveryTrigger`, I think you should take a look and probably suggest the approach you just used for gdi05a.
Discovery on entering a footprint is what some of the later games used for some missions, and I don't know this mission and how it should work, so I'm leaving it to you.

**Comment by obrakmann** [comment]
I have tested both now (sorry, was really tired last night).

``` diff
diff --git a/mods/cnc/maps/gdi05a/gdi05a.lua b/mods/cnc/maps/gdi05a/gdi05a.lua
index beb5b8d..8440928 100644
--- a/mods/cnc/maps/gdi05a/gdi05a.lua
+++ b/mods/cnc/maps/gdi05a/gdi05a.lua
@@ -144,7 +144,7 @@ SetupWorld = function()
        Reinforcements.Reinforce(gdi, GdiApc, { GdiApcEntry.Location, GdiApcRallyPoint.Location }, DateTime.Seconds(1), function(actor) actor.Stance = "Defend" end)
        Reinforcements.Reinforce(gdi, GdiInfantry, { GdiInfantryEntry.Location, GdiInfantryRallyPoint.Location }, 15, function(actor) actor.Stance = "Defend" end)

-       Trigger.OnPlayerDiscovered(gdiBase, DiscoverGdiBase)
+       Trigger.OnDiscovered(GdiSilo1, DiscoverGdiBase)

        Utils.Do(Map.NamedActors, function(actor)
                if actor.Owner == nod and actor.HasProperty("StartBuildingRepairs") then
```

Regarding that other mission, I'll probably add a comment that he can use this new trigger, but it doesn't make much sense for him to use it before it's in bleed.

**Comment by penev92** [comment]
True, but still point him to your changes so he has them in mind.

File: OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs [hunk 2]
```diff
@@ -407,14 +407,14 @@ public void OnInfiltrated(Actor a, LuaFunction func)
 
 		[Desc("Call a function when this actor is discovered by an enemy (doesn't work for neutral actors). " +
 			"The callback function will be called as func(Actor self, Player discoverer)")]
```
Comment (penev92) [inline OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs]: You forgot to remove the second argument here

File: OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs [hunk 3]
```diff
@@ -407,14 +407,14 @@ public void OnInfiltrated(Actor a, LuaFunction func)
...
-		public void OnDiscovered(Actor a, Player discoverer, LuaFunction func)
+		public void OnDiscovered(Actor a, LuaFunction func)
 		{
 			GetScriptTriggers(a).RegisterCallback(Trigger.OnDiscovered, func, Context);
 		}
 
 		[Desc("Call a function when this player is discovered by an enemy (doesn't work for neutrals). " +
-			"The callback function will be called as func(Player discovered, Player discoverer)")]
-		public void OnPlayerDiscovered(Player discovered, Player discoverer, LuaFunction func)
+			"The callback function will be called as func(Actor self, Player discoverer)")]
```
Comment (penev92) [inline OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs]: The method signatures still states `Player discovered`

File: OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs (hunk 2 above)
Comment (obrakmann) [inline OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs:409]: No, I didn't. The callback function _will_ have that argument.  Again, the callback is _not_ the function below this line. The callback is actually the `func` argument, which will be stored in the `ScriptTriggers` trait and then called when the event happens (on line 335 of `ScriptTriggers.cs`).

Comment (penev92) [inline OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs:409]: Yeah I remembered what you said above and just deleted my comment :D

**Pull request merged**